assets/js/app/Instance.js

```diff
@@ -18,46 +18,57 @@ LocusZoom.Instance = function(id, datasource, layout, state) {
     this.initialized = false;
 
     this.id = id;
-    this.parent = LocusZoom;
     
     this.svg = null;
 
-    // The _panels property stores child panel instances
-    this._panels = {};
+    // The panels property stores child panel instances
+    this.panels = {};
     this.remap_promises = [];
+
+    // The layout is a serializable object used to describe the composition of the instance
+    this.layout = layout || JSON.parse(JSON.stringify(LocusZoom.DefaultLayout));
     
     // The state property stores any instance-wide parameters subject to change via user input
-    this.state = state || {
-        chr: 0,
-        start: 0,
-        end: 0
-    };
+    this.state = state || JSON.parse(JSON.stringify(LocusZoom.DefaultState));
     
-    // The view property contains parameters that define the physical space of the entire LocusZoom object
-    this.view = {
-        width: 0,
-        height: 0,
-        min_width: 0,
-        min_height: 0
-    };
-
     // LocusZoom.Data.Requester
     this.lzd = new LocusZoom.Data.Requester(datasource);
-    
+
+    // Initialize the layout (should this happen in initialize()?)
+    this.initializeLayout();
+
     return this;
   
 };
 
-// Set the view dimensions for this instance. If an SVG exists, update its dimensions
+LocusZoom.Instance.prototype.initializeLayout = function(){
+
+    // Set instance dimensions or fall back to default values
+    this.layout.width      = this.layout.width      || LocusZoom.DefaultLayout.width;
```

Comment:
Actually I guess this is related to my note one line 29. We can either merge the user supplied layout with the default layout once at initialization, or we can use the `user||default` pattern before each time we extract a value. Is there a strong reason to choose one over the other?